### PR TITLE
Sarcasm Detection Model using Machine Learning(NLP) 

### PR DESCRIPTION
**Pull request by KamakshiOjha**
# Pull Request for ML-Crate 💡

## Issue Title: Sarcasm Detection Model using Machine Learning(NLP) 

- **Info about the related issue (Aim of the project)** : This repository contains various deep learning models for detecting sarcasm in text data using TensorFlow and Keras. Each model explores different architectures and techniques to improve sarcasm detection performance.
- **Name:**  Kamakshi Ojha
- **Email ID for further communication**: kkamakshiojha@gmail.com
- **GitHub ID:** https://github.com/KamakshiOjha
- **Idenitfy yourself:** AI/ML practitioner

Closes: #661

### Describe the add-ons or changes you've made 📃

I have added a new folder for sarcasm detection using NLP which includes the following:

- **Images**: Added Images for the accuracy and loss for each model.
- **Dataset**: Included the dataset required for training and testing the sarcasm detection models.
- **Models**: Implemented 7 models for sarcasm detection using different architectures and techniques.
- **README.md**: Created a comprehensive README file explaining the project, models.
- **requirements.txt**: Added a requirements file to specify the necessary Python packages for the project.

## Type of change ☑️

What sort of change have you made:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

Describe how it has been tested
To test the sarcasm detection models, we used a custom input headline and processed it through a predefined preprocessing function. 

Describe how have you verified the changes made
The raw prediction value and the classification (sarcastic or not) are printed to verify the model's performance as output.

## Checklist: ☑️
- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by abhisheks008 — APPROVED** [review]
Looks good to me. Approved @KamakshiOjha 

Need some time to merge this pull request as I haven't gone through the guidelines for adding the points for VSOC. I'll merge this by EOD or by Friday evening.
